modules/gltf/extensions/audio/gltf_document_extension_audio.cpp

```diff
@@ -0,0 +1,645 @@
+		Dictionary state_extensions = state_json["extensions"];
+		if (state_extensions.has("KHR_audio_emitter")) {
+			// KHR_audio_emitter's data is all defined in the document-level
+			// extensions and is designed to be highly reusable.
+			Dictionary khr_audio_ext = state_extensions["KHR_audio_emitter"];
+			Array audio_sources;
+			if (khr_audio_ext.has("sources")) {
+				ERR_FAIL_COND_V_MSG(!khr_audio_ext.has("audio"), ERR_PARSE_ERROR, "GLTF audio: KHR_audio_emitter sources are defined but there is no audio data.");
+				audio_sources = khr_audio_ext["sources"];
+				Array audio_data = khr_audio_ext["audio"];
+				for (int i = 0; i < audio_sources.size(); i++) {
+					Dictionary audio_source_dict = audio_sources[i];
+					int audio_data_index = audio_source_dict["audio"];
+					ERR_FAIL_INDEX_V_MSG(audio_data_index, audio_data.size(), ERR_PARSE_ERROR, "GLTF audio: KHR_audio_emitter source audio data index " + itos(audio_data_index) + " is not in the KHR_audio_emitter audio data array (size=" + itos(audio_data.size()) + ").");
+					Dictionary audio_data_dict = audio_data[audio_data_index];
+				}
+				p_state->set_additional_data(StringName("GLTFAudioData"), audio_data);
+				p_state->set_additional_data(StringName("GLTFAudioSources"), audio_sources);
+			}
+			if (khr_audio_ext.has("emitters")) {
+				Array audio_emitter_dicts = khr_audio_ext["emitters"];
+				Array audio_emitters;
+				for (int emitter_index = 0; emitter_index < audio_emitter_dicts.size(); emitter_index++) {
+					Ref<GLTFAudioPlayer> audio_emitter = GLTFAudioPlayer::from_dictionary(audio_emitter_dicts[emitter_index]);
+					audio_emitters.append(audio_emitter);
+				}
+				p_state->set_additional_data(StringName("GLTFAudioEmitters"), audio_emitters);
+			}
+		}
+	}
+	return OK;
+}
+
+Vector<String> GLTFDocumentExtensionAudio::get_supported_extensions() {
+	Vector<String> ret;
+	ret.push_back("KHR_audio_emitter");
+	return ret;
+}
+
+Error GLTFDocumentExtensionAudio::parse_node_extensions(Ref<GLTFState> p_state, Ref<GLTFNode> p_gltf_node, Dictionary &p_extensions) {
+	if (p_extensions.has("KHR_audio_emitter")) {
+		Dictionary khr_audio_ext = p_extensions["KHR_audio_emitter"];
+		if (khr_audio_ext.has("emitter")) {
+			int emitter_index = khr_audio_ext["emitter"];
+			p_gltf_node->set_additional_data(StringName("GLTFAudioEmitterIndex"), emitter_index);
+		}
+	}
+	return OK;
+}
+
+// Helper methods for import_post_parse.
+String _determine_mime_type_for_audio_dict(const Dictionary &p_audio_data_dict) {
+	String mime_type = p_audio_data_dict.get("mimeType", "");
+	if (!mime_type.is_empty()) {
+		return mime_type;
+	}
+	// Determine the MIME type from the URI.
+	String uri = p_audio_data_dict.get("uri", "");
+	if (uri.begins_with("data:")) {
+		return uri.substr(5, uri.find(";") - 5);
+	} else if (uri.ends_with(".mp3")) {
+		return "audio/mpeg";
+	} else if (uri.ends_with(".wav")) {
+		return "audio/wav";
+	} else if (uri.ends_with(".ogg")) {
+		return "audio/ogg";
```

Comment:
The decision, at least for now, is that the spec only requires support for the MP3 and WAV audio MIME types. Supporting Ogg Vorbis here is a superset of the spec. There is no harm in having this here, but I wanted to mention it. Additionally, with this PR, Ogg Vorbis cannot be exported, so Godot will only ever export valid to-spec files.